tests/test_geom_opt.py

```diff
@@ -120,3 +121,40 @@ def test_missing_traj_kwarg(tmp_path):
     traj_path = tmp_path / "NaCl-traj.xyz"
     with pytest.raises(ValueError):
         optimize(single_point.struct, traj_kwargs={"filename": traj_path})
+
+
+def test_hydrostatic_strain():
+    """Test setting hydrostatic strain for filter."""
+    single_point_1 = SinglePoint(
+        struct_path="./tests/data/NaCl-deformed.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+
+    single_point_2 = SinglePoint(
+        struct_path="./tests/data/NaCl-deformed.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+
+    atoms_1 = optimize(
+        single_point_1.struct, filter_kwargs={"hydrostatic_strain": True}
+    )
+    atoms_2 = optimize(
+        single_point_2.struct, filter_kwargs={"hydrostatic_strain": False}
+    )
+
+    expected_1 = [5.69139709, 5.69139709, 5.69139709, 89.0, 90.0, 90.0]
+    expected_2 = [5.68834069, 5.68893345, 5.68932555, 89.75938298, 90.0, 90.0]
```

Comment:
What is the tolerance on `pytest.approx`? May want to truncate these to match, otherwise it's somewhat misleading.

Reply:
It fails if you swap them the test fails - it's within a relative tolerance of `1e-6`.

The test is most significant for the first angle (fourth element), which is very comfortably outside the tolerance, but actually I think the first four elements all trigger the error.

Reply:
The next question is... do we care about them being within `1e-6`? Or does it just have to have a reasonable fit? Imagine in the future we change default optimiser. This will fail, but it's not physically unreasonable for them to differ by a small amount (certainly not 10^-6 Ang or 10^-6 deg), may `1e-2`, `1e-3`?

Reply:
I will say is good if changing the optimiser the test fails.

Reply:
Not necessarily, no, but I don't think it's necessarily possible to choose a tolerance that can both distinguish between optimising the same structure in slightly different ways while also being flexible enough to change optimization method completely.

E.g. with `rel=10e-4` only the angle is triggered as an error.

Reply:
This new structure is an improvement over the normal NaCl.cif file, but in general I think we need to redo the tests more broadly to distinguish results for different options better before we can have a more reasonable tolerance, and even then I wouldn't assume it would transfer to new optimizers.

Reply:
One issue that could be problematic is that we do indeed already change the default optimizer if you have updated ASE (as we recommend), so while the CI etc. passes with the tagged ASE, it could fail for people developing with newer versions.